docs/source/tutorials/psij.ipynb

```diff
@@ -0,0 +1,547 @@
+    "\n",
+    "environment = \"local\"\n",
+    "executable = \"/bin/date\"\n",
+    "ex = JobExecutor.get_instance(environment)\n",
+    "job = Job(JobSpec(executable=executable))\n",
+    "\n",
+    "print(\"Job Status Stages:\")\n",
+    "print(\"\\tPrior to submitting: \", str(job.status))\n",
+    "ex.submit(job)\n",
+    "print(\"\\tAfter submitting: \", str(job.status))\n",
+    "time.sleep(1)\n",
+    "print(\"\\tOnce completed: \", str(job.status))"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Multiple Jobs\n",
+    "This section will show you how to submit *multiple* jobs"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from psij import Job, JobExecutor, JobSpec\n",
+    "from psij import JobState\n",
+    "\n",
+    "executable = \"/bin/date\"\n",
+    "ex = JobExecutor.get_instance(environment)\n",
+    "\n",
+    "jobs = []\n",
+    "\n",
+    "print(\"Submitting jobs: \")\n",
+    "for _ in range(10):\n",
+    "    job = Job(JobSpec(executable=executable))\n",
+    "    print(str(job.status))\n",
+    "    jobs.append(job)\n",
+    "    ex.submit(job)\n",
+    "\n",
+    "time.sleep(1)\n",
+    "print(\"\\nCompleted Jobs: \")\n",
+    "_ = [print(str(job.status)) for job in jobs]"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "Every instance of `JobExecutor` can handle an abitrar number of submitted jobs. For reference, we have tested it with up to 64,000 jobs."
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "### Waiting for Completion\n",
+    "To wait for a job to complete once it has been submitted, it suffices to call the [wait](https://exaworks.org/psij-python/#docs/.generated/psij.html/#psij.job.Job.wait) method with no arguments:"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "from psij import Job, JobSpec\n",
+    "\n",
+    "job = Job(JobSpec(executable=\"/bin/date\"))\n",
+    "ex.submit(job)\n",
+    "job.wait()"
+   ]
+  },
+  {
+   "attachments": {},
+   "cell_type": "markdown",
+   "metadata": {},
+   "source": [
+    "The [wait](https://exaworks.org/psij-python/#docs/.generated/psij.html/#psij.job.Job.wait) call will return once the job has reached a terminal state, which almost always means that it finished or was cancelled.\n",
+    "\n",
+    "To distinguish jobs that complete successfully from ones that fail or are cancelled, fetch the status of the job after calling [wait](https://exaworks.org/psij-python/#docs/.generated/psij.html/#psij.job.Job.wait):"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": null,
+   "metadata": {},
+   "outputs": [],
+   "source": [
+    "job.wait()\n",
+    "print(job.status.state)"
```

Comment:
This may sound surprising, but in concurrent systems ordering is a tricky thing. I cannot say for certain that the print statement will always print the state that caused `job.wait()` to complete. However, `job.wait()` is guaranteed to return the state that made it complete. Hence, you could say:

```
    status = job.wait()
    print(status.state)
```

Reply:
This comment has been addressed